ci/ci/github.py

```diff
@@ -776,6 +777,10 @@ async def _start_deploy(self, batch_client, steps=()):
 (cd {shq(repo_dir)}; {self.checkout_script()})
 '''
             )
+            if sha_must_be:
+                out, err = sync_check_shell_output(f'(cd {repo_dir}; git rev-parse {self.sha})')
+                current_sha = out.decode().strip()
+                assert current_sha == sha_must_be, (current_sha, sha_must_be)
```

Comment:
Can we return a proper error here? I'd recommend using `assert` only for violated internal invariants.

Reply:
I was hoping that the whole thing shouldn't really happen, and we could remove the whole check after we figure out why the revisions don't match. But yeah, agree!

Reply:
Added a proper error